src/azure-cli-core/azure/cli/core/credential.py

```diff
@@ -0,0 +1,127 @@
+# --------------------------------------------------------------------------------------------
```

Comment:
Where does code in this file come from? I've seen similar code in cli core.

Reply:
The code is migrated from [adal_authentication.py](https://github.com/Azure/azure-cli/blob/d07ee662c8cb49978901fc4dc11ddad46c89c27a/src/azure-cli-core/azure/cli/core/adal_authentication.py).